book/embeds.md

```diff
@@ -1021,9 +1021,30 @@ Most of the layout tree's `paint` methods don't need to change, but to
 paint an `IframeLayout`, we'll need to paint the child frame:
 
 ``` {.python}
+def paint_tree(layout_object, display_list):
+    cmds = layout_object.paint()
+
+    if isinstance(layout_object, IframeLayout) and \
+        layout_object.node.frame:
+        paint_tree(layout_object.node.frame.document, cmds)
```

Comment:
Why do this in `paint_tree` instead of in `IframeLayout`?

Reply:
Because I wanted there to be exactly three concepts:

* `paint`, which paints the current layout object
* `paint_effect`, which paints effects applying to the whole subtree (including subframes)
* `paint_tree`, which walks the whole tree across frames and patches together calls to `paint` and `paint_effect` in the right order

Reply:
I guess that's fine. I don't have a strong theory of where exactly per-layout-object and paint-generic operations go.